File: frontend/src/types/chatbot.ts
// Chatbot types matching the FastAPI backend schemas

export interface ChatMessage {
  role: 'user' | 'assistant'
  content: string
}

export interface ChatSource {
  source: string
  snippet: string
}

export interface ChatQueryRequest {
  query: string
  conversation_history: ChatMessage[]
  max_history_turns?: number
}

export interface ChatQueryResponse {
  response: string
  sources_used: ChatSource[]
  regime_context_used: boolean
}

export interface ChatContextResponse {
  current_regime: string | null
  regime_confidence: number | null
  probabilities: Record<string, number>
  sentiment_scores: Record<string, number>
  stress_level: number | null
  vix_level: number | null
  volatility_regime: string | null
  volatility_score: number | null
  recent_transitions: Array<{
    from: string
    to: string
    date: string
    ciss: number | null
    vix: number | null
  }>
  summary: string
}

File: src/sentiment_detector/api/schemas/chatbot.py
"""Chatbot request/response schemas."""

from pydantic import BaseModel, Field


class ChatMessage(BaseModel):
    role: str = Field(..., pattern="^(user|assistant)$")
    content: str


class ChatQueryRequest(BaseModel):
    query: str = Field(..., min_length=1, max_length=2000)
    conversation_history: list[ChatMessage] = Field(default_factory=list)
    max_history_turns: int = Field(default=6, ge=0, le=20)


class ChatSource(BaseModel):
    source: str
    snippet: str


class ChatQueryResponse(BaseModel):
    response: str
    sources_used: list[ChatSource]
    regime_context_used: bool


class ChatContextResponse(BaseModel):
    """Live dashboard context for the chatbot system prompt."""

    current_regime: str | None
    regime_confidence: float | None
    probabilities: dict[str, float] = {}
    sentiment_scores: dict[str, float]
    stress_level: float | None
    vix_level: float | None = None
    volatility_regime: str | None = None
    volatility_score: float | None = None
    recent_transitions: list[dict]
    summary: str
